tests/v2/validation/provisioning/rke2/encryption_key_rotation_test.go

```diff
@@ -0,0 +1,270 @@
+			require.NoError(r.T(), err)
+
+			clusterName := namegen.AppendRandomString(fmt.Sprintf("%s-%s", r.clusterName, provider.Name))
+			generatedPoolName := fmt.Sprintf("nc-%s-pool1-", clusterName)
+			machinePoolConfig := provider.MachinePoolFunc(generatedPoolName, namespace)
+
+			machineConfigResp, err := testSessionClient.Steve.SteveType(provider.MachineConfigPoolResourceSteveType).Create(machinePoolConfig)
+			require.NoError(r.T(), err)
+
+			machinePools := machinepools.RKEMachinePoolSetup(nodesAndRoles, machineConfigResp)
+
+			cluster := clusters.NewK3SRKE2ClusterConfig(clusterName, namespace, "calico", credential.ID, kubeVersion, machinePools)
+
+			if strings.Contains(kubeVersion, "k3s") {
+				cluster.Spec.RKEConfig.MachineGlobalConfig.Data["secrets-encryption"] = true
+			}
+
+			clusterResp, err := clusters.CreateK3SRKE2Cluster(testSessionClient, cluster)
+			require.NoError(r.T(), err)
+
+			kubeProvisioningClient, err := r.client.GetKubeAPIProvisioningClient()
+			require.NoError(r.T(), err)
+
+			result, err := kubeProvisioningClient.Clusters(namespace).Watch(context.TODO(), metav1.ListOptions{
+				FieldSelector:  "metadata.name=" + cluster.ObjectMeta.Name,
+				TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+			})
+			require.NoError(r.T(), err)
+
+			checkFunc := clusters.IsProvisioningClusterReady
+
+			err = wait.WatchWait(result, checkFunc)
+			require.NoError(r.T(), err)
+			assert.Equal(r.T(), clusterName, clusterResp.ObjectMeta.Name)
+
+			require.NoError(r.T(), r.rotateEncryptionKeys(clusterResp.ID, 1, defaults.WatchTimeoutSeconds))
+			r.T().Logf("Successfully completed encryption key rotation for %s", name)
+
+			r.T().Logf("Creating %d secrets in namespace default for encryption key rotation for %s", scale, name)
+
+			clusterID, err := clusters.GetClusterIDByName(r.client, clusterName)
+			require.NoError(r.T(), err)
+			secretResource, err := kubeapi.ResourceForClient(r.client, clusterID, "default", secrets.SecretGroupVersionResource)
+			require.NoError(r.T(), err)
+
+			for i := 0; i < scale; i++ {
+				secret := &corev1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						GenerateName: fmt.Sprintf("encryption-key-rotation-test-%d-", i),
+					},
+					Data: map[string][]byte{
+						"key": []byte(namegen.RandStringLower(5)),
+					},
+				}
+				_, err = secrets.CreateSecret(secretResource, secret)
+				require.NoError(r.T(), err)
+			}
+
+			r.T().Logf("Successfully created %d secrets in namespace default for encryption key rotation for %s", scale, name)
+			// encryption key rotation is capped at 5 secrets per second (10 every 2 seconds), so 10000 secrets will take
+			// 2000 seconds which is ~33 minutes.
+			require.NoError(r.T(), r.rotateEncryptionKeys(clusterResp.ID, 2, 60*60))
+			r.T().Logf("Successfully completed second encryption key rotation for %s", name)
+		})
+	})
+}
+
+func (r *RKE2EncryptionKeyRotationTestSuite) rotateEncryptionKeys(id string, generation, timeout int64) error {
```

Comment:
I see we're passing `timeout` here, but how come you're only using it when watching `RKEControlPlanes` and using `defaults.WatchTimeoutSeconds` when watching `Clusters`? 

Also, this is probably a nitpick, but should `timeout` here be of type `time.Duration` rather than `int64`?

Reply:
From apimachinery:
```
	TimeoutSeconds *int64 `json:"timeoutSeconds,omitempty" protobuf:"varint,5,opt,name=timeoutSeconds"`
```